### PR TITLE
Remove all code deprecated in Rails 5.2

### PR DESCRIPTION
**Pull request by rafaelfranca**
@kaspth I need review on the commits with the title "Remove deprecated `config.secret_token`" and "
Remove secret_token rack env and cookie upgrade code"

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb [hunk 1]
```diff
@@ -76,22 +76,11 @@ class SQLite3Adapter < AbstractAdapter
...
-      #   ExampleModel.where("boolean_column = 't'").update_all(boolean_column: 1)
-      #   ExampleModel.where("boolean_column = 'f'").update_all(boolean_column: 0)
-      # for all models and all boolean columns, after which the flag must be set
-      # to true by adding the following to your <tt>application.rb</tt> file:
-      #
-      #   Rails.application.config.active_record.sqlite3.represent_boolean_as_integer = true
-      class_attribute :represent_boolean_as_integer, default: false
+      def self.represent_boolean_as_integer=(value) # :nodoc:
+        ActiveSupport::Deprecation.warn(
+          "`.represent_boolean_as_integer=` is deprecated and will be removed in Rails 6.1"
```
Comment (matthewd) [inline activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb]: Perhaps this should warn if `value` is true (maybe with a message more like "is now always true, so the setting is deprecated and will be removed"?), but raise if it's false?

**Review by kaspth — APPROVED** [review]
Those 2 commits look good to me!

**Review by kaspth — APPROVED** [review]
Now I checked the rest of them. Looks good!

File: railties/lib/rails/commands/server/server_command.rb [hunk 2]
```diff
@@ -275,22 +259,6 @@ def prepare_restart
           FileUtils.rm_f(options[:pid]) if options[:restart]
         end
 
-        def deprecate_positional_rack_server_and_rewrite_to_option(original_options)
-          if using
-            ActiveSupport::Deprecation.warn(<<~MSG)
-              Passing the Rack server name as a regular argument is deprecated
```
Comment (y-yagi) [inline railties/lib/rails/commands/server/server_command.rb]: This deprecation added by Rails 6.0. Ref: #32058

Comment (y-yagi) [reply railties/lib/rails/commands/server/server_command.rb]: I reverted fa791fb8e2a718b5d0430c7ca5a454678dfc192d with eb63faaa1af7ef28ae1a716d068acc447e28c174.

Comment (rafaelfranca) [reply railties/lib/rails/commands/server/server_command.rb]: Thank you for catching that.

**Comment by bogdanvlviv** [comment]
I've found two deprecations that should be removed as well:

- https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/routing/route_set.rb#L593-L605, those deprecations are in Rails since 5.1, see #23980

- https://github.com/rails/rails/blob/master/activerecord/lib/active_record/attribute_methods.rb#L194-L204

/cc @rafaelfranca

**Comment by rafaelfranca** [comment]
I left them by design.

**Pull request merged**